container/go/cmd/puller/puller.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This binary pulls images from a Docker Registry.
+// Unlike regular docker pull, the format this package uses is proprietary.
+
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+)
+
+var (
+	imgName         = flag.String("name", "", "The name location including repo and digest/tag of the docker image to pull and save. Supports fully-qualified tag or digest references.")
+	directory       = flag.String("directory", "", "Where to save the images files.")
+	clientConfigDir = flag.String("client-config-dir", "", "Specifies where the custom docker config.json is located. Overiddes the value from DOCKER_CONFIG.")
+	cachePath       = flag.String("cache", "", "Image's files cache directory.")
+	arch            = flag.String("architecture", "", "Image platform's CPU architecture.")
```

Comment:
You can refer to https://github.com/google/containerregistry/blob/master/tools/platform_args_.py for the platform docs. And be consistent with one stype for all your flags.